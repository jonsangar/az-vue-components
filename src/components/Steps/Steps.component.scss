$color-background: #ffffff;

.az-steps {
  overflow: hidden;
  background-color: $color-background;

  &--horizontal {
    padding: 10px 10px 0;

    .az-steps__items {
      position: relative;
      display: flex;
      align-items: center;
      justify-content: space-between;
      padding-bottom: 22px;
    }
  }

  &__line{
    z-index: 0;
    height: 1px;
    margin: 0 half();
    overflow: hidden;
    background-color: color(gray5);
    transform: translateY(-55px);
  }

  &--vertical {
    padding: 0 0 0 double();
  }
}
