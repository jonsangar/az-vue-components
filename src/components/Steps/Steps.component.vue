<template lang="pug">
  .az-steps--horizontal(
  :active="active"
  activeColor="var(color--success)")
    .az-steps__items
      slot
    .az-steps__line


</template>

<script>
export default {

  data() {

    return {
      active: 0
    }

  }
  // props: {
  //   activeColor: {
  //     type: String,
  //     default: 'red',
  //   },
  //   inactiveIcon: {
  //     type: String,
  //     default: 'dot'
  //   },
  //   inactiveColor: {
  //     type: String,
  //     default: '#777777'
  //   },
  //   activeIcon: {
  //     type: String,
  //     default: 'checked'
  //   },
  // }

}
</script>

<style lang="scss" src="./Steps.component.scss" scoped></style>
