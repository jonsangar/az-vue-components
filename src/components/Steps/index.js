import Steps from './Steps.component.vue';

export default Steps;
