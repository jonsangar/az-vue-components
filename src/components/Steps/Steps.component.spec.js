import { shallowMount, createLocalVue } from '@vue/test-utils';
// Importamos librerÃ­as de estilos e iconos
import SvgIcon from 'vue-svgicon'
import Steps from './Steps.component.vue';

const localVue = createLocalVue()
localVue.use(SvgIcon, {
  tagName: 'az-icon',
  classPrefix: 'az',
})

describe('Steps.vue', () => {
  const build = () => {
    const wrapper = shallowMount(Steps, {
      localVue,
    })

    return {
      wrapper,
    }
  }

  it('Comprueba renderizado del componente Steps con snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })
});
