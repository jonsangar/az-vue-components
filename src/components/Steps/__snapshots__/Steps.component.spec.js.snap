// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps.vue Comprueba renderizado del componente Steps con snapshot 1`] = `
<div active="0" activecolor="var(color--success)" class="az-steps--horizontal">
  <div class="az-steps__items"></div>
  <div class="az-steps__line"></div>
</div>
`;
